src/app/_models/grade_book.ts

```diff
@@ -0,0 +1,12 @@
+export type GradeBook = {
+    grade: number
```

Comment:
OPTIONAL: it would be nice to rename this to `tokens_received` for the sake of clarity.